Перезалив проекта 2

PavelRomashin opened the pull request:
Перезалив проекта 2

PavelRomashin committed:
Перезалив проекта 2

faritowich commented on app/src/main/java/com/example/playlistmaker/MainActivity.kt:
```diff
@@ -1,13 +1,55 @@
 
+        val search = findViewById<Button>(R.id.search)
+        val library = findViewById<Button>(R.id.library)
+        val settings = findViewById<Button>(R.id.settings)
```
Лучше давать такие названия, чтобы было понятно, что это именно кнопки, например, searchButton. Элементов на экране может быть очень много и иначе можно будет запутаться

faritowich commented on app/src/main/java/com/example/playlistmaker/MainActivity.kt:
```diff
@@ -1,13 +1,55 @@
+        settings.setOnClickListener {
+            val settingsIntent = Intent(this, SettingsActivity::class.java)
+            startActivity(settingsIntent)
+        }
```
В качестве практики сделать разные способы создания лисенеров хорошо, главное не забыть их потом убрать.

faritowich commented on app/src/main/res/drawable/search.xml:
```diff
@@ -0,0 +1,9 @@
+    android:viewportHeight="19">
+  <path
+      android:pathData="M13.579,11.95H12.721L12.417,11.656C13.481,10.418 14.122,8.81 14.122,7.061C14.122,3.161 10.961,0 7.061,0C3.161,0 0,3.161 0,7.061C0,10.961 3.161,14.122 7.061,14.122C8.81,14.122 10.418,13.481 11.656,12.417L11.95,12.721V13.579L17.381,19L19,17.381L13.579,11.95ZM7.061,11.95C4.356,11.95 2.173,9.766 2.173,7.061C2.173,4.356 4.356,2.173 7.061,2.173C9.766,2.173 11.95,4.356 11.95,7.06 ...
+      android:fillColor="#1A1B22"/>
```
этот цвет встречается неоднократно, его можно вынести в ресурсы

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,130 @@
+        app:iconTint="@color/buttonTextColor" />
+
+
+    <LinearLayout
```
Можно использовать FrameLayout, в котором задать gravity текста к началу лейаута, а иконки - к концу

PavelRomashin replied:
Вы имеете ввиду лучше строить экран, где 2 вертикальных массива - один с текстом, а второй с кнопками? Или строки через фреймы делать? второй вариант не срабатывает 
![image](https://user-images.githubusercontent.com/111704930/205454957-1e8a8704-aa51-4bd6-97d5-74cc7dda4841.png)
Или замечание относится к заголовку?)

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,130 @@
+
+        <androidx.appcompat.widget.SwitchCompat
+            android:layout_width="match_parent"
+            android:layout_height="61dp"
```
Обычно рекомендуется делать размеры кратными 8 (или 4). Это обеспечивает единообразие дизайна, а также связано с тем, что большинство размеров экрана делятся на 8. Иногда эту рекомендацию приходится нарушать, но лучше по возможности ей следовать.
Также см. https://material.io/design/layout/spacing-methods.html#baseline-grid

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,130 @@
+        android:paddingVertical="34dp"
+        android:paddingStart="17dp"
+        android:paddingEnd="15dp"
+        android:paddingBottom="28dp">
```
Вместо paddingStart и android:paddingEnd лучше задать paddingHorizontal.

Кроме того, в этом файле и в файле activity_main никакие величины не вынесены в ресурсы (размеры лейаутов и текста). Это стоит поправить, чтобы работа была принята.

Также я только после отправки работы на доработку обратил внимание, что файл activity_main не попал в этот pull request:( В целом по нему основное замечание то же самое - величины не вынесены в ресурсы.

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,130 @@
+            <TextView
+                android:layout_width="wrap_content"
+                android:layout_height="match_parent"
+                android:layout_weight="99"
```
Если заменить родительский LinearLayout на FrameLayout, то можно будет избавиться от этого костыля

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,130 @@
+            android:layout_width="match_parent"
+            android:layout_height="61dp"
+            android:paddingTop="21dp"
+            android:paddingBottom="21dp"
```
можно заменить на paddingVertical

PavelRomashin committed:
Исправление замечаний:
1) Убраны избыточные заглушки для разных типов слушателей
2) Создан файл ресурсов dimens, куда вынесены размеры
3) Для строк меню настроек Linearlayout заменены на FrameLayout
4) Большая часть размеров теперь кратны 4, остальные - согласно макету в фигме
5) Исправлены отступы(Padding).

PavelRomashin committed:
Исправление замечаний:
1) Убрана обводка кнопок в меню настроек
2) Убрано дублирующеся название приложения
3) Названия кнопок написаны теперь не заглавными буквами, а согласно макету.
4) Цвет кнопок в меню настроек изменен на серый
